api-gen/rendering/index.ts

```diff
@@ -1,19 +1,29 @@
 import {readFileSync, writeFileSync} from 'fs';
 import path from 'path';
 import {DocEntry} from './entities';
-import {transformAndRenderEntry} from './rendering';
+import {getRenderable} from './processing';
+import {renderEntry} from './rendering';
 
 /** The JSON data file format for extracted API reference info. */
-interface EntryList {
+interface EntryCollection {
+  moduleName: string;
   entries: DocEntry[];
 }
 
-/** Aggregate all JSON data source files into an array of docs entries. */
-function aggregateEntries(srcs: string[]): DocEntry[] {
-  return srcs.reduce((entries: DocEntry[], jsonDataFilePath: string) => {
+/** Parse all JSON data source files into an array of collections. */
+function parseEntryData(srcs: string[]): EntryCollection[] {
+  return srcs.map((jsonDataFilePath) => {
     const fileContent = readFileSync(jsonDataFilePath, {encoding: 'utf8'});
-    return entries.concat((JSON.parse(fileContent) as EntryList).entries) as DocEntry[];
-  }, []);
+    return JSON.parse(fileContent) as EntryCollection;
+  });
+}
+
+/** Gets a normalized filename for a doc entry. */
+function getNormalizedFilename(moduleName: string, entryName: string): string {
+  // Angular entry points all contain an "@" character, which we want to remove
+  // from the filename. We also swap `/` with an underscore.
+  const normalizedModuleName = moduleName.replace('@', '').replace('/', '_');
```

Comment:
You probably want to use `replaceAll` (at least for the slashes-if we are generating entries for deeper entry-points)

Reply:
Good catch, changed to a regex with the `g` flag (since `replaceAll` seems to not be in the active node types)